lib/ansible/modules/cloud/azure/azure_rm_aci.py

```diff
@@ -0,0 +1,471 @@
+
+        ip_address = None
+
+        if self.ip_address is not None:
+            # get list of ports
+            if self.ports:
+                ports = []
+                for port in self.ports:
+                    ports.append(Port(port, "TCP"))
+                ip_address = IpAddress(ports, self.ip_address)
+
+        containers = []
+
+        for container_def in self.containers:
+            name = container_def.get("name")
+            image = container_def.get("image")
+            memory = container_def.get("memory", 1.5)
+            cpu = container_def.get("cpu", 1)
+            ports = []
+
+            port_list = container_def.get("ports")
+            if port_list:
+                for port in port_list:
+                    ports.append(ContainerPort(port))
+
+            containers.append(Container(name, image, ResourceRequirements(ResourceRequests(memory, cpu)), None, ports))
+
+        parameters = ContainerGroup(location=self.location,
+                                    tags=self.tags,
+                                    containers=containers,
+                                    image_registry_credentials=registry_credentials,
+                                    restart_policy=None,
+                                    ip_address=ip_address,
+                                    os_type=self.os_type,
+                                    volumes=None)
+
+        try:
+            response = self.mgmt_client.container_groups.create_or_update(self.resource_group, self.name, parameters)
+
+        except CloudError as exc:
+            self.log('Error attempting to create the container instance.')
+            self.fail("Error creating the ACI instance: {0}".format(str(exc)))
+        return create_aci_dict(response)
+
+    def delete_aci(self):
+        '''
+        Deletes the specified container group instance in the specified subscription and resource group.
+
+        :return: True
+        '''
+        self.log("Deleting the ACI instance {0}".format(self.name))
+        try:
+            response = self.mgmt_client.container_groups.delete(self.resource_group, self.name)
+        except CloudError as e:
+            self.log('Error attempting to delete the ACI instance.')
+            self.fail("Error deleting the ACI instance: {0}".format(str(e)))
+
+        return True
+
+    def get_aci(self):
+        '''
+        Gets the properties of the specified container service.
+
+        :return: deserialized ACI instance state dictionary
+        '''
+        self.log("Checking if the ACI instance {0} is present".format(self.name))
+        found = False
+        try:
+            response = self.mgmt_client.container_groups.get(self.resource_group, self.name)
+            found = True
+            self.log("Response : {0}".format(response))
+            self.log("ACI instance : {0} found".format(response.name))
+        except CloudError as e:
+            self.log('Did not find the ACI instance.')
```

Comment:
should check `e.error.error == 'ResourceNotFound'` before assuming that was the error